src/benchmarks/micro/libraries/System.Utf8String.Experimental/Perf.Utf8String.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Text.Experimental
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.Runtime)]
+    public class Perf
+    {
+        public string ascii_11;
+        public Utf8String ascii_11_ustring;
+        public string nonascii_110;
+        public Utf8String nonascii_110_ustring;
+        public string nonascii_chinese;
+        public Utf8String nonascii_chinese_ustring;
+        public string nonascii_cyrillic;
+        public Utf8String nonascii_cyrillic_ustring;
+        public string nonascii_greek;
+        public Utf8String nonascii_greek_ustring;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            string path = Path.Combine(Environment.CurrentDirectory, "libraries", "System.Utf8String.Experimental");
+            ascii_11 = File.ReadAllText(Path.Combine(path, "11.txt"));
+            ascii_11_ustring = new Utf8String(ascii_11);
+            nonascii_110 = File.ReadAllText(Path.Combine(path, "11-0.txt"));
+            nonascii_110_ustring = new Utf8String(nonascii_110);
+            nonascii_chinese = File.ReadAllText(Path.Combine(path, "25249-0.txt"));
+            nonascii_chinese_ustring = new Utf8String(nonascii_chinese);
+            nonascii_cyrillic = File.ReadAllText(Path.Combine(path, "30774-0.txt"));
+            nonascii_cyrillic_ustring = new Utf8String(nonascii_cyrillic);
+            nonascii_greek = File.ReadAllText(Path.Combine(path, "39251-0.txt"));
+            nonascii_greek_ustring = new Utf8String(nonascii_greek);
+        }
+
+        [Benchmark]
+        public int ToUtf16_nonascii_110()
+        {
+            string str = nonascii_110;
+            Utf8Span span = new Utf8Span(nonascii_110_ustring);
```

Comment:
Can we just put everything except last line in `[GlobalSetup]` so we want be allocating / triggering GC in every run and the measurements will be accurate?

Reply:
That's the first thing I tried :) The problem is that Utf8Span is a ref struct. I did check that the constructor for `Utf8Span` is cheap though. I don't see another way to initialize a `Span` here. @GrabYourPitchforks, do you think the `Utf8Span` constructor will skew our numbers here? From what I can tell it just get a reference to the first byte in the string and it's all on the stack, so I would expect it to be incredibly cheap. For reference, I'm trying to benchmark the `Utf8Utility.TranscodeToUtf16` method here (which internally calls both `ASCIIUtility.WidenAsciiToUtf16` and `ASCIIUtility.WidenFourAsciiBytesToUtf16AndWriteToBuffer` )

Reply:
I don't think it should matter. `ToChars()` just calls static `Utf8.Utf16()` method and return. So you can pass the same `utf8Span()` again and again. 
Ref: https://source.dot.net/#System.Private.CoreLib/Utf8Span.Conversion.cs,103